src/book/02-system/01-foundations/07-director.mdx

```diff
@@ -0,0 +1,447 @@
+
+Without `Needs`, the Provider will still run and can emit subtasks, but the subtasks will not do anything if another Provider has access to them.
+
+### When
+
+A When condition is used to ensure that a Provider will only execute when the specified condition is met.
+These conditions are provided using a global enum message along with a statement that describes what condition you want.
+For example, if you wanted a kick engine to only run if the robot is statically stable or better then you would write:
+
+```cpp
+on<Provide<Kick>, When<Stability, std::greater_equal, Stability::STANDING>>().then(...);
+```
+
+These conditions are updated by using normal NUClear emits. For example if
+
+```cpp
+emit(std::make_unique<Stability>(Stability::STATIC));
+```
+
+was emitted in another reaction, then the `When` condition would be met and the Provider would run.
+
+### Causing
+
+A Causing statement is a promise that by running this Provider, a specified desired state will be eventually reached.
+This is used to allow smooth transitions between modules.
+For example, when the walk engine stops moving it would be in a `STANDING` Stability state.
+This means the walk engine could have a `Provide` reaction that causes this state:
+
+```cpp
+on<Provide<Walk>, Causing<Stability, Stability::STANDING>().then({
+    // Make the robot stop walking
+    set_orders = Eigen::Vector3d::Zero();
+    // Check if we have stopped moving
+    if (state == STOPPED) {
+        // Emit that we are standing
+        emit(std::make_unique<Stability>(Stability::STANDING));
+    }
+});
+```
+
+The implementation for the reaction should result in the desired state being eventually emitted. This would not be the primary Provider for the Walk, but would instead be actie when a higher priority Provider wants this state.
+
+For example, if the kick engine only runs `When<Stability, std::greater_equal, Stability::STANDING>` then it would not be able to run while the walk engine is running normally. However, if the kick Task has a higher priority than the walk Task, then it will push the walk engine into the Causing Provide reaction, which will then cause the Stability state to be STANDING so that the kick engine can run.
+
+## Tasks
+
+Tasks are the jobs that Providers execute. The are generally in the form of Protobuf messages, and may or may not contain data within them.
+
+Tasks are implemented using NUClears `emit` along with a custom scope `emit<Task>`.
+
+```cpp
+// Emit a Task requesting to walk forwards
+emit<Task>(std::make_unique<Walk>(Eigen::Vector2d(1.0, 0.0, 0.0)));
+```
+
+This walk engine Task contains information on the speed and direction that the robot should walk in. Other Tasks, such as `GetUpWhenFallen` does not need any data.
+
+Tasks can take four arguments
+
+```cpp
+emit<Task>(data, priority, optional, name)
+```
+
+where
+
+- data: A pointer to a Protobuf message.
```

Comment:
I think this might look nicer as a table maybe... Or at least put each of the params in backticks?